### ISSUE TITLE
Ability to set a theme as default value

### ISSUE THREAD

**Issue by amirrezaask** [nvim-telescope/telescope.nvim#848]
**Is your feature request related to a problem? Please describe.**
I want to set a theme ( for example ivy ) as the default theme in telescope.setup

**Describe the solution you'd like**
require("telescope").setup({theme = 'ivy' })

**Comment by medwatt** [comment]
> **Is your feature request related to a problem? Please describe.**
> I want to set a theme ( for example ivy ) as the default theme in telescope.setup
> 
> **Describe the solution you'd like**
> require("telescope").setup({theme = 'ivy' })

Why don't you create a new function, set it to use the theme you want, and possibly add options different from the default (the rest of the options will be inherited from the default), bind the function to some key or command. 

```
local open_dir = function(config)
    local opt = require('telescope.themes').get_ivy {
        cwd = config.dir,
        prompt_title = config.title,
        shorten_path = true
    }
    require('telescope.builtin').find_files(opt)
end

```

**Comment by amirrezaask** [comment]
Thank you for tour help, i created a small function to wrap my calls to telescope and add theme options to it so i think i dont need a builtin thing yet i think it would help if such feature exists

**Comment by ok-nick** [comment]
I ran across this issue and wondered why there still isn't an option for setting the default theme? We have them for layouts and numerous other configurations, it would only make sense to add one for themes?

Maybe this issue should be reopened?

**Comment by rolflobker** [comment]
Just ran into same, couldn't figure out why setting as default didn't take and was wondering if I really had to set it for each individual picker. Turns out I do.

I appreciate very often keys and functions will be assigned, in which case this option can be passed. That still requires each picker to be set-up individually

I'd like it if any time, any ":Telescope somepicker" was ran it would use a default theme.

For now I am going to do pickers = { <picker>={theme="ivy")) for each picker available

Would make config cleaner if we could do { defaults  = {theme="ivy}}

**Comment by Conni2461** [comment]
yeah this issue should have been reopened, we need to find a suitable solution for this issue.

**Comment by rolflobker** [comment]
After snooping around in the source files, turns out the themes set layout options which can be set as default. I was unaware of this. Doing it this way does greatly reduces config.

This did the trick for me:

defaults = {
          layout_strategy = 'bottom_pane',
          layout_config = {
                height = 0.4,
          },
          border = true,
          sorting_strategy = "ascending",
[..]
}

Still would probably make more sense to allow setting a 'theme' in defaults which would result in even "cleaner" code:

defaults = {
          theme = "ivy",
[..]
}

The latter just "feels right"

**Comment by bartvanraaij** [comment]
@rolflobker thanks for pointing that out.

Instead of copy-pasting the theme's options, you could also import it into your defaults table, like so: 

```lua
telescope.setup({
telescope.setup({
  defaults = vim.tbl_extend(
      "force",
      require('telescope.themes').get_dropdown(), 
      {
        --- your own `default` options go here, e.g.:
        path_display = {
          truncate = 2
        },
        mappings = {
          i = {
            ["<esc>"] = actions.close,
          },
        }
      }
  ),
  extensions = {},
  [...]
})
```

This way you don't have to update your config if a the upstream theme is changed.

I do however fully agree that it would be so much better to be able to set
```
telescope.setup({
 defaults = {
  theme = "dropdown", -- Or whatever theme you like
},
```

**Comment by Rydwxz** [comment]
I found it a bit annoying that no default config is possible but I have a much easier workaround. This is an exceprt from my kmp.lua, which contains all my keymaps and at the top of which kym is already aliased to vim.keymap.set()
```lua
--telescope
local ivy = require('telescope.themes').get_ivy()
kym('n', '<leader>ff', function() require('telescope.builtin').find_files(ivy) end )
kym('n', '<leader>fs', function() require('telescope.builtin').live_grep(ivy) end )--string cwd
kym('n', '<leader>ft', function() require('telescope.builtin').treesitter(ivy) end )--find symbol
...
```
As you can see only a single line is added to the config, using the keybindings you already have.

**Comment by tudorjnu** [comment]
I came up with a similar solution but because I personally do not like the clutter of making an anonymous function inline, I made a wrapper:

```lua
local function theme_wrapper(telescope_command)
  return function()
    telescope_command(require("telescope.themes").get_dropdown())
  end
end
```

then things become:
```lua
{ "<leader>sh", theme_wrapper(builtin.help_tags), desc = "Help Pages" }
```

**Comment by rodrigo-sys** [comment]
```lua
require('telescope').setup({
  defaults = require('telescope.themes').get_dropdown(),
})
```

**Comment by tudorjnu** [comment]
I've tried the solution above. There is a weird bug when I am using it. This is the code that I am using:

```lua

defaults = require("telescope.themes").get_dropdown({
  preview = false,
  mappings = {
    n = {
      ["<A-p>"] = action_layout.toggle_preview,
    },
    i = {
      ["<A-p>"] = action_layout.toggle_preview,
    },
  },
}),
 
```

Now, if I have the preview set to false, I cannot toggle it. But if I have it unset, I can toggle it. If it is not wrapped, it works well.
